sw/device/tests/sysrst_ctrl_ulp_z3_wakeup_test.c

```diff
@@ -102,6 +106,18 @@ static void pinmux_setup(void) {
     CHECK_DIF_OK(dif_pinmux_output_select(&pinmux, kOutputPads[i],
                                           kPeripheralOutputs[i]));
   }
+  sysrst_ctrl_testutils_setup_dio(&pinmux);
+  sysrst_ctrl_testutils_release_dio(&sysrst_ctrl, true, true);
+}
+
+static uint8_t read_phase_pins(void) {
+  uint32_t gpio_state;
+  CHECK_DIF_OK(dif_gpio_read_all(&gpio, &gpio_state));
+  // Since the host may not be able to change all pins atomically, we use
+  // a Gray code encoding so that it suffices to change one pin to go to
+  // the next phase.
+  static const uint8_t kGrayCode[] = {0, 1, 3, 2, 7, 6, 4, 5};
```

Comment:
Note the jump between `2` and `7` is 2 bits, the graycode should be:

`static const uint8_t kGrayCode[] = {0, 1, 3, 2, 6, 7, 5, 4};`

to only flip one bit at a time

Reply:
This is for decoding the Gray code, not encoding, so it's the "reversed" mapping.

Reply:
Ah, I see. You might want to add a comment saying that it is an inverse look up, but the code it correct. Thanks!